### PR TITLE
insert link to leave reference in notification email

### LINKED ISSUE #1671: Add link to write reference in emails when other person writes a reference
When a host or surfer writes a reference, the other party gets an email. If that other party hasn't yet written a reference, we should include a direct link to write one.

### PR DESCRIPTION
**Pull request by sndrr**
.. to make it easier to leave feedback for other users and make Couchers grow with one less click.

Closes #1671

**Backend checklist**
- [ ] Formatted my code by running `autoflake -r -i --remove-all-unused-imports src && isort . && black .` in `app/backend`
- [ ] Added tests for any new code or added a regression test if fixing a bug
- [ ] All tests pass
- [ ] Run the backend locally and it works
- [ ] Added migrations if there are any database changes, rebased onto `develop` if necessary for linear migration history

**Web frontend checklist**
- [ ] Formatted my code with `yarn format && yarn lint --fix`
- [ ] There are no warnings from `yarn lint`
- [ ] There are no console warnings when running the app
- [ ] Added any new components to storybook
- [ ] Added tests where relevant
- [ ] All tests pass
- [ ] Clicked around my changes running locally and it works
- [ ] Checked Desktop, Mobile and Tablet screen sizes

### CODE REVIEW

**Comment by sndrr** [comment]
First issue, work in progress.

File: app/backend/templates/host_reference.md [hunk 1]
```diff
@@ -17,6 +17,8 @@ Thanks for using Couchers to organize this interaction! We hope you had an enjoy
 {% else %}
 Please go and write a reference for them too. It's a nice gesture and helps us build a community together! When you've both written a reference, both references will become visible. Otherwise {{ reference.from_user.name|couchers_escape }}'s reference will become visible 2 weeks after the end of your interaction, after which you cannot write a reference back.
 
+{{ button("Write a reference", leave_reference_link)|couchers_safe }} for {{ reference.from_user.name|couchers_escape}} and make Couchers a better place.
+
```
Comment (aapeliv) [inline app/backend/templates/host_reference.md:21]: I don't think the button works inline, it'll be rendered as a block like this

Screenshot from 2021-10-25 20-28-23

Also I don't think the button will work in plaintext emails. Maybe something like this?

```
{% if html %}
{{ button("Write a reference", leave_reference_link)|couchers_safe }}

Alternatively, click the following link: {{ link(leave_reference_link, html)|couchers_safe }}.
{% else %}
<{{ leave_reference_link|couchers_escape }}>
{% endif %}
```

**Review by aapeliv — CHANGES_REQUESTED** [review]
Sorry for the slow review. Besides the small comment, I think this is good to go!

**Comment by aapeliv** [comment]
Continued at #2308

**Pull request closed without merging**